src/hooks/usePublisher.ts

```diff
@@ -0,0 +1,71 @@
+import { useEffect, useRef, useState } from "react";
+import { Director, Publish } from '@millicast/sdk';
+
+export type PublisherState = "ready" | "connecting" | "streaming";
+
+export interface Publisher {
+    startStreaming: (broadcastOptions: BroadcastOptions) => Promise<void>;
+    stopStreaming: () => void;
+    updateAudioTrack: (mediaStream: MediaStream) => Promise<void>;
+    updateVideoTrack: (mediaStream: MediaStream) => Promise<void>;
+    publisherState: PublisherState;
+}
+
+export interface BroadcastOptions {
+    mediaStream: MediaStream
+    // ADD More
+}
+
+const usePublisher = (token: string, streamName: string): Publisher => {
+
+    const [publisherState, setPublisherState] = useState<PublisherState>("ready");
+    const publisher = useRef<Publish>();
+
+    useEffect(() => {
+        console.log(token, streamName);
```

Comment:
Please delete this console log, thanks.
Instead, you can inspect it by opening the source code actually :)